components/generator/imagevector/src/main/kotlin/io/github/composegears/valkyrie/generator/imagevector/spec/BackingPropertySpec.kt

```diff
@@ -0,0 +1,115 @@
+package io.github.composegears.valkyrie.generator.imagevector.spec
+
+import androidx.compose.material.icons.generator.ClassNames
+import androidx.compose.material.icons.generator.vector.Vector
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.FunSpec
+import com.squareup.kotlinpoet.PropertySpec
+import com.squareup.kotlinpoet.buildCodeBlock
+import io.github.composegears.valkyrie.generator.ext.fileSpecBuilder
+import io.github.composegears.valkyrie.generator.ext.getterFunSpecBuilder
+import io.github.composegears.valkyrie.generator.ext.propertySpecBuilder
+import io.github.composegears.valkyrie.generator.ext.removeDeadCode
+import io.github.composegears.valkyrie.generator.ext.setIndent
+import io.github.composegears.valkyrie.generator.imagevector.ImageVectorSpecConfig
+import io.github.composegears.valkyrie.generator.imagevector.ImageVectorSpecOutput
+import io.github.composegears.valkyrie.generator.imagevector.util.backingPropertyName
+import io.github.composegears.valkyrie.generator.imagevector.util.backingPropertySpec
+import io.github.composegears.valkyrie.generator.imagevector.util.iconPreviewSpec
+import io.github.composegears.valkyrie.generator.imagevector.util.iconPreviewSpecForNestedPack
+
+internal class BackingPropertySpec(
+    private val config: ImageVectorSpecConfig,
+) {
+
+    fun createAsBackingProperty(vector: Vector): ImageVectorSpecOutput {
+        val backingProperty = backingPropertySpec(
+            name = config.iconName.backingPropertyName(),
+            type = ClassNames.ImageVector,
+        )
+
+        val iconPackClassName = when {
+            config.iconPack.isEmpty() -> null
+            else -> {
+                if (config.iconNestedPack.isEmpty()) {
+                    ClassName(
+                        config.iconPackage,
+                        config.iconPack,
+                    )
+                } else {
+                    ClassName(
+                        config.iconPackage,
+                        config.iconPack,
+                    ).nestedClass(config.iconNestedPack)
+                }
+            }
+        }
+
+        val packageName = config.resolvePackageName()
```

Comment:
Still duplicate.

Reply:
Most logic of these 2 specs could be reused.

Reply:
reuse more code, other looks similar but has differences. I would like to leave readability as it just a 2 options for generation kotlin code 🙂